File: miniprogram/api/handlers/food.go
package handlers

import (
	"github.com/gin-gonic/gin"
	"github.com/mini-program/api/config"
	"github.com/mini-program/api/utils"
	"net/http"
)

// GetFoodList 获取菜品列表
func GetFoodList(c *gin.Context) {
	var foods []config.Food
	if err := config.DB.Find(&foods).Error; err != nil {
		utils.Error(c, http.StatusInternalServerError, utils.ErrorCodeInternalError, "获取菜品列表失败")
		return
	}

	utils.Success(c, foods)
}

// AddFood 添加菜品
func AddFood(c *gin.Context) {
	var food config.Food
	if err := c.ShouldBindJSON(&food); err != nil {
		utils.Error(c, http.StatusBadRequest, utils.ErrorCodeParamInvalid, "无效的请求数据")
		return
	}

	// 检查菜品名称是否已存在
	var existingFood config.Food
	if err := config.DB.Where("name = ?", food.Name).First(&existingFood).Error; err == nil {
		utils.Error(c, http.StatusBadRequest, utils.ErrorCodeDuplicate, "该菜品已存在")
		return
	}

	if err := config.DB.Create(&food).Error; err != nil {
		utils.Error(c, http.StatusInternalServerError, utils.ErrorCodeInternalError, "添加菜品失败")
		return
	}

	utils.Success(c, food)
}

// DeleteFood 删除菜品
func DeleteFood(c *gin.Context) {
	id := c.Param("id")
	
	var food config.Food
	if err := config.DB.First(&food, id).Error; err != nil {
		utils.Error(c, http.StatusNotFound, utils.ErrorCodeNotFound, "菜品不存在")
		return
	}

	if err := config.DB.Delete(&food).Error; err != nil {
		utils.Error(c, http.StatusInternalServerError, utils.ErrorCodeInternalError, "删除菜品失败")
		return
	}

	utils.Success(c, gin.H{"message": "删除成功"})
} 

File: miniprogram/services/food.js
import api from './api'

const foodService = {
  // 获取菜品列表
  async getFoodList() {
    try {
      const result = await api.get('/foods');
      if (result && result.data && Array.isArray(result.data)) {
        // 转换字段名为小写
        return result.data.map(item => ({
          id: item.ID,
          name: item.Name
        }));
      }
      return [];
    } catch (error) {
      console.error('获取菜品列表失败:', error);
      return [];
    }
  },

  // 添加菜品
  async addFood(food) {
    try {
      const response = await api.addFood(food)
      return { success: true, data: response.data }
    } catch (error) {
      console.error('添加菜品失败:', error)
      return { 
        success: false, 
        error: error.message,
        errorCode: error.errorCode
      }
    }
  },

  // 删除菜品
  async deleteFood(id) {
    try {
      const response = await api.deleteFood(id)
      return { success: true, data: response.data }
    } catch (error) {
      console.error('删除菜品失败:', error)
      return { 
        success: false, 
        error: error.message,
        errorCode: error.errorCode
      }
    }
  },

  // 随机选择菜品
  async getRandomFood() {
    try {
      const foodList = await this.getFoodList()
      if (foodList.length === 0) return null
      const randomIndex = Math.floor(Math.random() * foodList.length)
      const randomFood = foodList[randomIndex]
      return {
        id: randomFood.ID || randomFood.id,
        name: randomFood.Name || randomFood.name
      }
    } catch (error) {
      console.error('随机选择菜品失败:', error)
      throw {
        errorCode: error.errorCode || -1,
        message: error.message || '随机选择菜品失败'
      }
    }
  }
}

export default foodService 